apps/enc.c

```diff
@@ -293,6 +296,12 @@ int enc_main(int argc, char **argv)
             iter = opt_int_arg();
             pbkdf2 = 1;
             break;
+        case OPT_SALTLEN:
+            if (!opt_int(opt_arg(), &saltlen))
+                goto opthelp;
+            if (saltlen > (int)sizeof(salt))
+                saltlen = (int)sizeof(salt);
```

Comment:
Raising an error would be better than silently not doing what was asked.

Reply:
I thought about doing that, but it already behaves in strange ways when you specify the -S option... i.e. It truncates it to 8 or zero pads it to 8